### PR TITLE
mgr/devicehealth: replace CLICommand with CLIReadCommand

### PR DESCRIPTION
**Pull request by noviicee**
Fixes: https://tracker.ceph.com/issues/51631
Signed-off-by: Anamika <cseanamika3@gmail.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by noviicee** [comment]
@varshar16 please check this one.

**Comment by varshar16** [comment]
Hi Anamika, please see the submitting patches [doc](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst) and update the patch

**Comment by varshar16** [comment]
Also checkout the methods implementation in [mgr_module.py file](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/mgr_module.py#L314-L434).

**Comment by noviicee** [comment]
Okay! Up to it.

**Comment by noviicee** [comment]
> Also checkout the methods implementation in [mgr_module.py file](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/mgr_module.py#L314-L434).

What exactly is needed to be done here? :sweat-smile:

**Comment by noviicee** [comment]
> Also checkout the methods implementation in [mgr_module.py file](https://github.com/ceph/ceph/blob/master/src/pybind/mgr/mgr_module.py#L314-L434).

What exactly is needed to be done here? :sweat-smile:

File: src/pybind/mgr/devicehealth/module.py [hunk 1]
```diff
@@ -165,7 +165,7 @@ def is_valid_daemon_name(self, who: str) -> bool:
             return False
         return parts[0] in ('osd', 'mon')
 
-    @CLICommand('device query-daemon-health-metrics',
+    @CLIReadCommand('device query-daemon-health-metrics',
                 perm='r')
```
Comment (varshar16) [inline src/pybind/mgr/devicehealth/module.py:169]: If it a read only command then you replace like this
```suggestion
    @CLIReadCommand('device query-daemon-health-metrics')
```

File: src/pybind/mgr/devicehealth/module.py [hunk 2]
```diff
@@ -215,15 +215,15 @@ def do_get_health_metrics(self, devid: str, sample: Optional[str] = None) -> Tup
         return self.show_device_metrics(devid, sample)
 
     @CLIRequiresDB
-    @CLICommand('device check-health',
+    @CLIWriteCommand('device check-health',
                 perm='rw')
```
Comment (varshar16) [inline src/pybind/mgr/devicehealth/module.py:219]: This change is wrong. See `CLIWriteCommand` is for write only command https://github.com/ceph/ceph/blob/84be9581dc0ac9f064d5095708eb5d15d506178d/src/pybind/mgr/mgr_module.py#L433-L434

Comment (varshar16) [reply src/pybind/mgr/devicehealth/module.py:219]: This needs to be
```suggestion
    @CLICommand('device check-health')
```

File: src/pybind/mgr/devicehealth/module.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 import errno
 import json
-from mgr_module import MgrModule, CommandResult, CLIRequiresDB, CLICommand, Option
```
Comment (varshar16) [inline src/pybind/mgr/devicehealth/module.py:7]: Don't remove `CLICommand` and check if `CLIWriteCommand` is required

File: src/pybind/mgr/devicehealth/module.py (hunk 1 above)
Comment (noviicee) [reply src/pybind/mgr/devicehealth/module.py:169]: Understood!

Comment (noviicee) [reply src/pybind/mgr/devicehealth/module.py:169]: Understood!

Comment (noviicee) [reply src/pybind/mgr/devicehealth/module.py:169]: Understood!

File: src/pybind/mgr/devicehealth/module.py (hunk 2 above)
Comment (noviicee) [reply src/pybind/mgr/devicehealth/module.py:219]: Okay

File: src/pybind/mgr/devicehealth/module.py (hunk 3 above)
Comment (noviicee) [reply src/pybind/mgr/devicehealth/module.py:7]: Okay 👍

Comment (noviicee) [reply src/pybind/mgr/devicehealth/module.py:7]: `CLIWriteCommand` is not required so removing it from import.

**Review by varshar16 — CHANGES_REQUESTED** [review]
The changes look good. Please merge the commits into a single commit.

**Comment by varshar16** [comment]
@noviicee please do rebase: `git rebase origin/master`

**Comment by noviicee** [comment]
Hi @varshar16 this one got too messed up. Can I open a new PR?

**Comment by varshar16** [comment]
Also don't forget to sign your patch and add tracker ticket.
``
Fixes: https://tracker.ceph.com/issues/51631
Signed-off-by: Anamika cseanamika3@gmail.com
```

**Comment by noviicee** [comment]
> @noviicee please do rebase: `git rebase origin/master`and see this [doc](https://www.git-tower.com/learn/git/faq/git-squash/) on squashing commits.

I did the exact same thing. Don't know how this got so messed up. I am sorry. 
I think I should open a new PR.

**Comment by varshar16** [comment]
> > @noviicee please do rebase: `git rebase origin/master`and see this [doc](https://www.git-tower.com/learn/git/faq/git-squash/) on squashing commits.
> 
> I did the exact same thing. Don't know how this got so messed up. I am sorry.
> I think I should open a new PR.

Don't open a new PR. Did you rebase?

**Comment by noviicee** [comment]
> > > @noviicee please do rebase: `git rebase origin/master`and see this [doc](https://www.git-tower.com/learn/git/faq/git-squash/) on squashing commits.
> > 
> > 
> > I did the exact same thing. Don't know how this got so messed up. I am sorry.
> > I think I should open a new PR.
> 
> Don't open a new PR. Did you rebase?

Yes

**Comment by varshar16** [comment]
> > > > @noviicee please do rebase: `git rebase origin/master`and see this [doc](https://www.git-tower.com/learn/git/faq/git-squash/) on squashing commits.
> > > 
> > > 
> > > I did the exact same thing. Don't know how this got so messed up. I am sorry.
> > > I think I should open a new PR.
> > 
> > 
> > Don't open a new PR. Did you rebase?
> 
> Yes

No issues, share the output of this `git remote -v`

**Comment by ideepika** [comment]
@noviicee just missing <> in your mail id: in commit ba10e6292edda2ba606954d040a7509bf967288b

1. you can use git commit --amend 
2. add <> around mail <cseanamika3@gmail.com> and save changes
3. force push amended changes using `git push -f`

**Comment by noviicee** [comment]
> @noviicee just missing <> in your mail id: in commit [ba10e62](https://github.com/ceph/ceph/commit/ba10e6292edda2ba606954d040a7509bf967288b)
> 
> 1. you can use git commit --amend
> 2. add <> around mail [cseanamika3@gmail.com](mailto:cseanamika3@gmail.com) and save changes
> 3. force push amended changes using `git push -f`

Done!

**Comment by varshar16** [comment]
jenkins test make check

**Review by varshar16 — APPROVED** [review]
Congratulations!! Thank you for contributing to ceph.

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-07-17_04:37:01-rados-wip-kefu-testing-2021-07-17-1006-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-07-17_15:53:41-rados-wip-kefu-testing-2021-07-17-1518-distro-basic-smithi/

one of the failures is tracked by:

- https://tracker.ceph.com/issues/50245

the e2e failures are not related.

**Pull request merged**